### PR TITLE
Allow infinite integration span

### PR DESCRIPTION
**Pull request by jvalet**
Hello,
This pull request aims at answering #225, which itself originates from an Orekit need expressed in [issue #1005](https://gitlab.orekit.org/orekit/orekit/-/issues/1005).

### CODE REVIEW

**Comment by maisonobe** [comment]
This change seems OK to me.
Could you add two basic tests (one for the double version, one for the field version)?

**Comment by jvalet** [comment]
Hi Luc, thank you for the review.
I have added tests for EmbeddedRungeKuttaIntegratorAbstractTest and EmbeddedRungeKuttaFieldIntegratorAbstractTest.
I could also add them for the remaining integrators if it is required.

File: hipparchus-ode/src/test/java/org/hipparchus/ode/nonstiff/EmbeddedRungeKuttaFieldIntegratorAbstractTest.java [hunk 1]
```diff
@@ -678,6 +680,26 @@ public void testIssue118() {
...
+                    public Decimal64 g(FieldODEStateAndDerivative<Decimal64> state) {
+                        return state.getTime().subtract(pb.getFinalTime());
+                    }
+                    @Override
+                    public Action eventOccurred(FieldODEStateAndDerivative<Decimal64> state, boolean increasing) {
+                        return Action.STOP;
+                    }
+                },
+                Double.POSITIVE_INFINITY, 1.0e-6, 1000);
+        fieldIntegrator.integrate(new FieldExpandableODE<>(pb), pb.getInitialState(), Decimal64.POSITIVE_INFINITY);
```
Comment (maisonobe) [inline hipparchus-ode/src/test/java/org/hipparchus/ode/nonstiff/EmbeddedRungeKuttaFieldIntegratorAbstractTest.java:700]: Almost there!
Just add a check that integration stops at the expected final time.
Currently, the test does not check anything.
Same applies to the regular double version.

Comment (jvalet) [reply hipparchus-ode/src/test/java/org/hipparchus/ode/nonstiff/EmbeddedRungeKuttaFieldIntegratorAbstractTest.java]: Many thanks for the feedback!
I suppose the tests check the absence of failures... But I do see your point.
I added the check for both tests. Sorry for the omision.

**Pull request merged**